src/emitters/spot.cpp

```diff
@@ -0,0 +1,176 @@
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/warp.h>
+#include <mitsuba/render/emitter.h>
+#include <mitsuba/render/medium.h>
+#include <mitsuba/render/texture.h>
+
+NAMESPACE_BEGIN(mitsuba)
+
+/**!
+
+.. _emitter-spot:
+
+Spot light source (:monosp:`spot`)
+------------------------------------
+
+.. pluginparameters::
+
+ * - intensity
+   - |spectrum|
+   - Specifies the maximum radiant intensity at the center in units of power per unit steradian. (Default: 1).
+ * - cutoff_angle
+   - |float|
+   - Cutoff angle, beyond which the spot light is completely black (Default: 20 degrees)
+ * - beam_width
+   - |float|
+   - Subtended angle of the central beam portion (Default: :math:`cutoff_angle \times 3/4`)
+ * - texture
+   - |texture|
+   - An optional texture to be projected along the spot light
```

Comment:
In Mitsuba 2 spectrum properties (like `intensity`) are can always be spatially-variant. Therefore this `texture` is redondant. Maybe we could remove it? Or would it break backward compatibility?

Reply:
This was there in Mitsuba 1 and so it would break backward compatibility indeed. Should it be removed?

Reply:
For now let's only keep `m_intensity` internally. In the constructor, if only `texture` if provided (check with `has_property()`) then you use it to set `m_intensity`. If both are provided, it should throw and error for now. 

In the future, we could for instance multiply those two textures, but it isn't possible to do for now.